extensions/BugModal/lib/WebService.pm

```diff
@@ -27,6 +27,25 @@ use List::MoreUtils qw(any first_value);
 
 sub rest_resources {
     return [
+        # return all the products accessible by the user.
+        # required by new-bug
+        qr{^/bug_modal/products}, {
+            GET => {
+                method => 'products'
+            },
+        },
+
+        # return all the components pertaining to the product.
+        # required by new-bug
+        qr{^/bug_modal/components}, {
+            GET => {
+                method => 'components',
+                params => sub {
+                    return { product_name => Bugzilla->input_params->{product} }
```

Comment:
wow, this is a surprising use of this! I don't think we've ever used it this way, but it is interesting.
Probably this can be empty and just use Bugzilla->input_params from the product sub.